internal/host/dhcpallocatecmd.go

```diff
@@ -0,0 +1,84 @@
+package host
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/network"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/assisted-service/models"
+)
+
+type dhcpAllocateCmd struct {
+	baseCmd
+	dhcpAllocateImage string
+	db                *gorm.DB
+}
+
+func NewDhcpAllocateCmd(log logrus.FieldLogger, dhcpAllocateImage string, db *gorm.DB) *dhcpAllocateCmd {
+	return &dhcpAllocateCmd{
+		baseCmd:           baseCmd{log: log},
+		dhcpAllocateImage: dhcpAllocateImage,
+		db:                db,
+	}
+}
+
+func asMAC(macStr string) *strfmt.MAC {
+	mac := strfmt.MAC(macStr)
+	return &mac
+}
+
+func (f *dhcpAllocateCmd) prepareParam(host *models.Host) (string, bool, error) {
+	var cluster common.Cluster
+	if err := f.db.Take(&cluster, "id = ?", host.ClusterID.String()).Error; err != nil {
+		return "", false, err
+	}
+	if !swag.BoolValue(cluster.VipDhcpAllocation) || cluster.MachineNetworkCidr == "" || host.Inventory == "" {
```

Comment:
Add documentation.
If only one of this parameters is missing that's fine not an error?

Reply:
It is OK that one of the parameters is missing.  It is not an error:
- if VipDhcpAllocation is false - DHCP mode is disabled (like today)
- MachineNetworkCidr is empty - Was not set by the user
- Inventory is empty - was not received from the host.

Reply:
OK. Please document it with a comment

Reply:
OK

Reply:
Moved to the caller and added comment